api/workAllocation/index.ts

```diff
@@ -63,8 +63,7 @@ export async function searchTask(req: EnhancedRequest, res: Response, next: Next
     res.status(status);
     // Assign actions to the tasks on the data from the API.
     if (data) {
-      const caseworkers: Caseworker[] = await retrieveAllCaseWorkers(req, res);
-      assignActionsToTasks(data.tasks, req.body.view, caseworkers);
+      assignActionsToTasks(data.tasks, req.body.view, req.session.caseworkers);
```

Comment:
I don't understand why this change has been made. The `await retrieveAllCaseWorkers(req, res)` call was already returning it from `req.session.caseworkers` if it was on there and, if not, it was loading the caseworkers.

Reply:
This is made because the normal caseworker who is logged in cannot access caseworkers api. Its easier to show rather than type

Reply:
If I am logged in as case worker the assignee name is not required.
Only if I am logged in as senior case worker the assignee name is required

Reply:
Where is that logic in the code?